.github/workflows/tests-e2e-upgrade.yaml

```diff
@@ -324,7 +324,7 @@ jobs:
           kubectl patch node kind-worker3 --type=json -p='[{"op":"add","path":"/metadata/labels/cilium.io~1no-schedule","value":"true"}]'
 
           if ${{ matrix.encryption == 'ipsec' }}; then
-            cilium encrypt create-key --auth-algo rfc4106-gcm-aes
+            cilium encrypt create-key --auth-algo ${{ matrix.key-one }}
```

Comment:
(not directly related to this line) 

I feel like the way to make these workflows more maintainable is to delete the ipsec-related parts and move that exclusively to a dedicated workflow for validating ipsec. This is doing the opposite, making this workflow even more complex.